gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/ComputeExpiredFilesUDF.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.iceberg;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.iceberg.GcMetadataUtil;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableMetadataParser;
+import org.apache.iceberg.io.FileIO;
+import org.apache.spark.sql.api.java.UDF1;
+
+class ComputeExpiredFilesUDF implements UDF1<String, List<String>> {
```

Comment:
I know. But there is manifest pruning optimization to avoid data files of the live snapshots. 
I will try to optimize it. If not, I will come back with a clear explanation of why it is present twice.